.github/workflows/ci.yaml

```diff
@@ -0,0 +1,176 @@
+name: CI
+on:
+  push: 
+    branches:
+      - develop
+      - release*
+    tags: [v*]
+    paths-ignore:
+      - README.md
+      - CHANGELOG.md
+      - LICENSE
+  pull_request:
+  # Sequence of patterns matched against refs/heads
+    branches:    
+      - develop
+      - release*
+    paths-ignore:
+      - README.md
+      - CHANGELOG.md
+      - LICENSE
+
+env:
+  project: 'focal-freedom-236620'
+  image: 'controller'
+
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+      packages: 'write'
+    name: Preflight
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+    - run: npm ci
+    - run: npm run standard
+    - run: |
+        npm i -g better-npm-audit
+        npx better-npm-audit audit -p
+
+  Tests:
+    needs: Build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+      packages: 'write'
+      issues: read
+      checks: write
+      pull-requests: write
+    strategy:
+      matrix:
+        node: [ 16, 17, 16, 19 ]
+    name: Node ${{ matrix.node }} Test
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup Node ${{ matrix.node }} Test
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node }}
+    - name: Cache Node Modules
+      uses: actions/cache@v3
+      with:
+        path: |
+          node_modules
+        key: ${{ runner.os }}-controller-node_modules-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-controller-node_modules-
+    - run: npm ci
+    - run: npm run test -- junit
+    - run: npm run postman_test
+    - name: Publish Test Results
+      uses: EnricoMi/publish-unit-test-result-action@v2
+      if: always()
+      with:
+        files: |
+          *-results.xml
+
+  Publish:
+    needs: [Build, Tests]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+      packages: 'write'
+    name: Publish Controller
+    if: ${{ needs.Tests.result }} == 'success'
```

Comment:
This line _might_ be redundant because of the `needs` above. But it is harmless either way.

Reply:
Removed!